sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -447,46 +459,55 @@ private static Map<String, TPrivate> createMapIEDNameAndPrivate(SclRootAdapter s
                 .map(TLNode::getPrivate).flatMap(Collection::stream)
                 .collect(Collectors.toList())
                 .stream()
-                .filter(tPrivate -> tPrivate.getType().equals(CommonConstants.COMPAS_ICDHEADER) && getValueFromPrivate(tPrivate, CommonConstants.IED_NAME) != null)
-                .collect(Collectors.toMap(tPrivate -> getValueFromPrivate(tPrivate, CommonConstants.IED_NAME), tPrivate -> tPrivate));
+                .filter(tPrivate ->
+                        tPrivate.getType().equals(CommonConstants.COMPAS_ICDHEADER)
+                                && getCompasICDHeader(tPrivate).isPresent() && getCompasICDHeader(tPrivate).get().getIEDName() != null)
+                .collect(Collectors.toMap(tPrivate -> getCompasICDHeader(tPrivate).get().getIEDName(), tPrivate -> tPrivate));
     }
 
     private static Map<String, Pair<TPrivate, List<SCL>>> createMapICDSystemVersionUuidAndSTDFile(Set<SCL> stds) {
         Map<String, Pair<TPrivate, List<SCL>>> stringSCLMap = new HashMap<>();
         stds.forEach(std -> std.getIED().forEach(ied -> ied.getPrivate().forEach(tp -> {
-            if (tp.getType().equals(CommonConstants.COMPAS_ICDHEADER)) {
-               String icdSysVer = getValueFromPrivate(tp, CommonConstants.ICD_SYSTEM_VERSION_UUID);
-               if(icdSysVer != null) {
-                   Pair<TPrivate, List<SCL>> pair = stringSCLMap.get(icdSysVer);
-                   List<SCL> list = pair != null ? pair.getRight() : new ArrayList<>();
-                   list.add(std);
-                   stringSCLMap.put(icdSysVer, Pair.of(tp,list));
-               }
+            Optional<TCompasICDHeader> optIcdHeader = getCompasICDHeader(tp);
+            if (optIcdHeader.isPresent() && optIcdHeader.get().getICDSystemVersionUUID() != null) {
+                String icdSysVer = optIcdHeader.get().getICDSystemVersionUUID();
+                Pair<TPrivate, List<SCL>> pair = stringSCLMap.get(icdSysVer);
+                List<SCL> list = pair != null ? pair.getRight() : new ArrayList<>();
```

Comment:
You could throw an Exception here if `pair != null`. This is related to [this comment](https://github.com/com-pas/compas-sct/pull/102/files#r885633091)